polar-core/src/kb.rs

```diff
@@ -574,7 +575,26 @@ impl KnowledgeBase {
             }
             .into());
         }
-        self.constants.insert(name, value);
+
+        if let Value::ExternalInstance(ExternalInstance {
+            class_id,
+            instance_id,
+            ..
+        }) = *value.value()
+        {
+            if class_id.map_or(false, |id| id == instance_id) {
```

Comment:
This logic seems a bit brittle. Probably better to have a different FFI for registering class ids. Not an issue for now.

Reply:
agreed! we are stretching things here and I think it'd make life easier in the other language impls to not have to piggyback atop `ExternalInstance`